pkg/metrics/memory.go

```diff
@@ -19,14 +19,16 @@ type SwapSample struct {
 }
 
 type MemorySample struct {
-	MemoryTotal       float64 `json:"memoryTotalBytes"`
-	MemoryFree        float64 `json:"memoryFreeBytes"`
-	MemoryUsed        float64 `json:"memoryUsedBytes"`
-	MemoryFreePercent float64 `json:"memoryFreePercent"`
-	MemoryUsedPercent float64 `json:"memoryUsedPercent"`
-	MemoryCachedBytes float64 `json:"memoryCachedBytes"`
-	MemorySlabBytes   float64 `json:"memorySlabBytes"`
-	MemorySharedBytes float64 `json:"memorySharedBytes"`
+	MemoryTotal       float64  `json:"memoryTotalBytes"`
+	MemoryFree        float64  `json:"memoryFreeBytes"`
+	MemoryUsed        float64  `json:"memoryUsedBytes"`
+	MemoryFreePercent float64  `json:"memoryFreePercent"`
+	MemoryUsedPercent float64  `json:"memoryUsedPercent"`
+	MemoryCachedBytes float64  `json:"memoryCachedBytes"`
+	MemorySlabBytes   float64  `json:"memorySlabBytes"`
+	MemorySharedBytes float64  `json:"memorySharedBytes"`
+	MemoryBuffers     *float64 `json:"memoryBuffers,omitempty"`
```

Comment:
I think we normally use pointers to be able to differentiate between not having a value and a `0` value
In this case I'm seeing that we always have the value, is there any reason for having this as pointer?

Reply:
We don't want to provide MemoryKernelFree for Windows as it has the same value as MemoryFree, so it would be a duplicated metric. MemoryBuffers is only available in Linux, thus in the other OS would be always 0.